opentimelineio/adapters/fcp_xml.py

```diff
@@ -210,7 +210,17 @@ def _parse_media_reference(file_e, element_map):
     url = file_e.find('./pathurl').text
     file_rate = _parse_rate(file_e, element_map)
     timecode_rate = _parse_rate(file_e.find('./timecode'), element_map)
-    timecode_frame = int(file_e.find('./timecode/frame').text)
+
+    frame_e = file_e.find('./timecode/frame')
+    if frame_e is not None:
+        timecode_frame = int(frame_e.text)
+
+    else:
+        timecode_frame = otio.opentime.from_timecode(
```

Comment:
Catching the two cases looks sound to me. According to the FCP7 XML spec timecode must contain either a string sub-element, or a frame sub-element. 
But perhaps it would make more sense to create RationalTime objects for each case?
```Python
if frame_e is None:
    start_time = otio.opentime.RationalTime(
        int(frame_e.text), timecode_rate
    )
else:
    start_time = otio.opentime.from_timecode(
        file_e.find('./timecode/string).text, timecode_rate
    )
```
Then later for available_range you can do
```Python
available_range = otio.opentime.TimeRange(
    start_time=start_time, 
    duration=otio.opentime.RationalTime(duration, file_rate
)
```

Reply:
Sure! Sounds good.
Should I do the same for the duration perhaps?

Reply:
Duration should be fine. I mainly mentioned the start_time because after otio.opentime.from_timecode, you already have a rationalTime object. So in that case you were creating the same object twice.